s2spy/time.py

```diff
@@ -304,3 +296,237 @@ def show(self) -> pd.DataFrame:
+        Returns:
+            The calendar mapped to the input start and end year.
+        """
+        if start > end:
+            raise ValueError("The start year cannot be greater than the end year")
+
+        self._first_year = start
+        self._last_year = end
+        self._mapping = "years"
+        return self
+
+    def _label_targets(self, intervals: pd.DataFrame) -> pd.DataFrame:
+        """Adds target labels to the header row of the intervals.
+
+        Args:
+            intervals (pd.Dataframe): Dataframe with intervals.
+
+        Returns:
+            pd.Dataframe: Dataframe with target periods labelled.
+        """
+        return intervals.rename(
+            columns={i: f"(target) {i}" for i in range(self.n_targets)}
+        )
+
+    def get_intervals(self) -> pd.DataFrame:
+        """Method to retrieve updated intervals from the Calendar object."""
+        if self._mapping is None:
+            raise ValueError(
+                "Cannot retrieve intervals without map_years or "
+                "map_to_data having configured the calendar."
+            )
+        # if self._mapping == "data":
+        #     self._set_year_range_from_timestamps()
+
+        year_range = range(
+            self._last_year, self._first_year - 1, -(self._get_skip_nyears() + 1)
+        )
+
+        intervals = pd.concat([self._map_year(year) for year in year_range], axis=1).T
+
+        intervals.index.name = "anchor_year"
+        return intervals
+
+    def _get_skip_nyears(self) -> int:
+        """Determine how many years need to be skipped to avoid overlapping data.
+
+        Required to prevent information leakage between anchor years.
+
+        Returns:
+            int: Number of years that need to be skipped.
+        """
+        years = pd.to_timedelta(
+            self._total_length_target + self._total_length_precursor
+        ) / pd.Timedelta("365days")
+
+        return 0 if self._allow_overlap else int(np.ceil(years).astype(int) - 1)
+
+    def show(self) -> pd.DataFrame:
+        """Displays the intervals the Calendar will generate for the current setup.
+
+        Returns:
+            pd.Dataframe: Dataframe containing the calendar intervals, with the target
+                periods labelled.
+        """
+        return self._label_targets(self.get_intervals())
+
+
+class Period:
+    """Basic construction element of calendar for defining target period."""
+
+    def __init__(self, length: int, gap: int = 0, target: bool = False) -> None:
+        self.length = length
+        self.gap = gap
+        self.target = target
+        # TO DO: support lead_time
+        # self.lead_time = lead_time
+
+
+def target_period(length, gap: int = 0):
+    """Instantiate a build block as target period. """
+    return Period(length=length, gap=gap, target=True)
+
+
+def precursor_period(length, gap: int = 0):
+    """Instantiate a build block as precursor period."""
+    return Period(length=length, gap=gap, target=False)
```

Comment:
This basically works as a drop-in replacement by the way, requiring no changes elsewhere ;)

Reply:
Thanks for the comment, Bart. Actually I thought about it at the beginning, but then felt that it might be an overkill in `time.py`...But I think the big advantage of this design is, we can be more flexible in case later on we might want to add something to Precursor Period or Target Period. I think we can put it this way.

Reply:
Right now it's not really needed functionally, but it does indeed add flexibility if we need it in the future. Additionally, it's more clear to the user that they are a class. Because the function `target_period()` returns a class `Period` already.

Reply:
That makes sense to me. Good point! 😉